src/spaced/spaced/game/weapons/gun_beam.cpp

```diff
@@ -0,0 +1,131 @@
+#include <bave/graphics/sprite.hpp>
+#include <bave/imgui/im_text.hpp>
+#include <spaced/game/weapons/gun_beam.hpp>
+
+namespace spaced {
+using bave::im_text;
+using bave::NotNull;
+using bave::Ptr;
+using bave::Rect;
+using bave::Rgba;
+using bave::Seconds;
+using bave::Shader;
+using bave::Sprite;
+
+namespace {
+class LaserCharge : public IWeaponRound {
+  public:
+	using Config = GunBeam::Config;
+
+	explicit LaserCharge(NotNull<ILayout const*> layout, Config config, glm::vec2 const muzzle_position)
+		: m_layout(layout), m_config(config), m_fire_remain(config.fire_duration) {
+		m_ray.transform.position.y = muzzle_position.y;
+	}
+
+	[[nodiscard]] auto is_destroyed() const -> bool final { return m_fire_remain <= 0s; }
```

Comment:
just for the sake of my curiosity, can you explain this syntax a bit? why write it this way as opposed to `bool is_destroyed() const { return m_fire_remain <= 0s; }` ?

Reply:
It's mostly a style thing, my reasons for preferring trailing return types:
- It matches the syntax for lambda expressions.
- It can often be less verbose, eg, `Foo::Bar Foo::get_bar()` vs `auto Foo::get_bar() -> Foo`.
- It's sometimes required, eg with a function template whose return type depends on the types of the parameters.
- All function names end up having identical indentation.

Reply:
got it, thanks for the explanations!